Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/LocationUpdateService.java

```diff
@@ -368,8 +374,159 @@ private void updateDriverStatus(Boolean status) {
+        String val = sharedPreff.getString(k, null);
+        return val;
+    }
+
+    private JSONObject getLatLng(Double lat , Double lng) throws JSONException {
+        JSONObject latLng = new JSONObject();
+        latLng.put("lat", lat);
+        latLng.put("lon", lng);
+        return latLng;
+    }
+
+    private void getRoute(Double startLat, Double startLng, Double endLat, Double endLng) {
+        StringBuilder result = new StringBuilder();
+
+        String regToken = getValueFromStorage("REGISTERATION_TOKEN");
+        String baseUrl = getValueFromStorage("BASE_URL");
+        String version = getValueFromStorage("VERSION_NAME");
+
+        if(regToken == null || baseUrl == null || version == null) {
+            return;
+        }
+
+        try {
+            String url = baseUrl + "/trip/route";
+            HttpURLConnection connection = (HttpURLConnection) (new URL(url).openConnection());
+            connection.setRequestMethod("POST");
+            connection.setRequestProperty("Content-Type", "application/json");
+            connection.setRequestProperty("token", regToken);
+            connection.setRequestProperty("x-client-version", version);
+
+            JSONObject payload = new JSONObject();
+
+            JSONObject startLatLng = getLatLng(startLat, startLng);
+            JSONObject endLatLng = getLatLng(endLat, endLng);
+
+            JSONArray waypoints = new JSONArray();
+            waypoints.put(startLatLng);
+            waypoints.put(endLatLng);
+
+            payload.put("waypoints", waypoints);
+            payload.put("mode", "CAR");
+            payload.put("calcPoints", true);
+
+            OutputStream stream = connection.getOutputStream();
+            stream.write(payload.toString().getBytes());
+            connection.connect();
+            int respCode = connection.getResponseCode();
+            InputStreamReader respReader;
+            if (respCode == 200) {
+                respReader = new InputStreamReader(connection.getInputStream());
+                BufferedReader in = new BufferedReader(respReader);
+                String inputLine;
+                while ((inputLine = in.readLine()) != null) {
+                    result.append(inputLine);
+                }
+                JSONObject res = (JSONObject) new JSONArray(String.valueOf(result)).get(0);
+
+                JSONArray pathLatLng = (JSONArray) res.get("points");
+
+                rideWaypoints = String.valueOf(pathLatLng);
+            } else {
+                Bundle params = new Bundle();
+                mFirebaseAnalytics.logEvent("LS_ERROR_GETTING_ROUTE", params);
+            }
+
+        } catch (Exception e) {
+            Log.e("get route api exception", e.toString());
+        }
+    }
+
+
+    private void updateDeviation(double latitude, double longitude, String waypoints) throws JSONException {
```

Comment:
If there's a try/catch block with the whole function. No need to throw JSONException off the function